tsl/src/continuous_aggs/create.c

```diff
@@ -1835,6 +1835,29 @@ tsl_process_continuous_agg_viewstmt(Node *node, const char *query_string, void *
 	timebucket_exprinfo = cagg_validate_query((Query *) stmt->into->viewQuery);
 	cagg_create(stmt, &viewstmt, (Query *) stmt->query, &timebucket_exprinfo, with_clause_options);
 
+	/* Refreshing unless requested to not refresh. Cannot use
+	 * DirectFunctionCall since we pass NULL parameters to function. */
+	if (!stmt->into->skipData)
+	{
+		LOCAL_FCINFO(fcinfo, 3);
+		Oid relid;
+
+		CommandCounterIncrement();
+
+		relid = get_relname_relid(stmt->into->rel->relname, nspid);
+
+		InitFunctionCallInfoData(*fcinfo, NULL, 3, InvalidOid, NULL, NULL);
+
+		FC_SET_ARG(fcinfo, 0, ObjectIdGetDatum(relid));
+		FC_SET_NULL(fcinfo, 1);
+		FC_SET_NULL(fcinfo, 2);
+
+		ereport(NOTICE,
+				(errmsg("refreshing continuous aggregate"),
+				 errhint("Use WITH NO DATA if you do not want to refresh the continuous aggregate "
+						 "on creation.")));
```

Comment:
Nit:

```suggestion
				(errmsg("refreshing continuous aggregate \"%s\"", get_rel_name(relid)),
				 errhint("Use WITH NO DATA if you do not want to refresh the continuous aggregate "
						 "on creation.")));
```

Reply:
This is in response to a statement that creates the view in question, so I do not think there is any additional value in mentioning the name of the object just being created.

Reply:
i think this message should be removed since this is an unconditional message that will always be shown unless the user specified with no data, so this is the expected behaviour and nothing the user needs to be informed of

Reply:
@mkindahl I think it is common practice to mention the object. For instance, when you create a table and one already exists:

```
postgres=# create table mytable (time int);
ERROR:  relation "mytable" already exists
```

making the message self-contained is useful since it might go into a log and might be read out-of-context. I guess you can log table statements too, but I think it is optional. You might also want to grep/search for errors relating to a specific object.

Another example, when creating a view (temporary) the following notice is raised:

```
	ereport(NOTICE,
				(errmsg("view \"%s\" will be a temporary view",
						view->relname)));
```

I also worry  that the message will look too much like a  left-over debug message (also Sven's concern) unless it is connected to a specific object and action.

Reply:
@svenklemm FWIW, I think there are other cases when we raise informational messages although they are about the default. For instance, we always add a NOT NULL constraint to the time column, and tell people about it. The same goes for default index creation, etc. 

I think refreshing a continuous aggregate on creation has special semantics, given that it might involve big data sets, and it might be useful for people to know why the view creation takes a long time.

Reply:
@erimatnor Ok. I'll add the object name since there are precedent.